lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -30,20 +31,24 @@
 
 <li class="course-item">
   % if settings.FEATURES.get('ENABLE_VERIFIED_CERTIFICATES'):
-    % if enrollment.mode == "verified":
-      % if verification_status.get('status') in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
-        <% mode_class = " verified" %>
-      % else:
-        <% mode_class = " honor" %>
-      % endif
-    % else:
-      <% mode_class = " " + enrollment.mode %>
-    % endif
+    <% course_verified_certs = CourseMode.get_certificate_display(enrollment.mode, verification_status.get('status')) %>
+    <%
+        display_mode = course_verified_certs.get('display_mode')
+        if display_mode == 'verified':
+             mode_class = 'verified'
+        elif display_mode == 'honor':
+            mode_class = 'honor'
+        elif display_mode == 'professional':
+            mode_class = 'professional'
+        else:
```

Comment:
I was doing this earlier but in your previous you mentioned let the html decide the css class so i added the logic here. I did't check test cases earlier was busy in other task i will fix the test case.